migration-guides.md

```diff
@@ -0,0 +1,128 @@
+# Migration guides
+
+> :star: Switching from other HTTP request libraries to Got :star:
+
+### Migrating from Request
+
+You may think it's too hard to switch, but it's really not. 🦄
+
+Let's take the very first example from Request's readme:
+
+```js
+const request = require('request');
+
+request('https://google.com', (error, response, body) => {
+	console.log('error:', error); // Print the error if one occurred
+	console.log('statusCode:', response && response.statusCode); // Print the response status code if a response was received
```

Comment:
I know this is from the Requests docs, but this makes no sense. Why would you guard `response`, but not `body`?...

Reply:
There is much sense! :D If `response` is `null`/`undefined` you can refer to `undefinded` by `response.statusCode`. It throws. But you can log `undefined`, right?

Reply:
Try:

```js
const a = undefined;
console.log(a.b);
console.log(a);
```

Reply:
AFAIK if there's an error the rest is `null`.

Reply:
You're right. I'm too used to Got. It's just the Request API that makes no sense for me in general.